lib/google/apis/version.rb

```diff
@@ -21,16 +23,19 @@ module Apis
     # @private
     OS_VERSION = begin
       if RUBY_PLATFORM =~ /mswin|win32|mingw|bccwin|cygwin/
-        `ver`.sub(/\s*\[Version\s*/, '/').sub(']', '')
+        output, _ = Open3.capture2('ver')
+        output.sub(/\s*\[Version\s*/, '/').sub(']', '')
       elsif RUBY_PLATFORM =~ /darwin/i
-        "Mac OS X/#{`sw_vers -productVersion`}"
+        output, _ = Open3.capture2('sw_vers -productVersion')
```

Comment:
If it contains whitespace, the single-argument form will still get passed to a shell. You can avoid the shell by splitting it into separate arguments, in which case it gets `exec`ed directly. The documentation mentions how that works: https://ruby-doc.org/core-2.7.1/Process.html#method-c-spawn
```suggestion
        output, _ = Open3.capture2('sw_vers', '-productVersion')
```